internal/provider/models/destinations/gcp_cloud_monitoring.go

```diff
@@ -0,0 +1,136 @@
+		"resource_labels": schema.MapAttribute{
+			Optional:    true,
+			ElementType: StringType{},
+			Description: "Key/Value pair used to describe the resource",
+			Validators: []validator.Map{
+				mapvalidator.All(
+					mapvalidator.KeysAre(stringvalidator.LengthAtLeast(1)),
+					mapvalidator.ValueStringsAre(stringvalidator.LengthAtLeast(1)),
+				),
+			},
+		},
+	}, nil),
+}
+
+func GcpCloudMonitoringDestinationFromModel(plan *GcpCloudMonitoringDestinationModel, previousState *GcpCloudMonitoringDestinationModel) (*Destination, diag.Diagnostics) {
+	dd := diag.Diagnostics{}
+
+	component := Destination{
+		BaseNode: BaseNode{
+			Type:        GCP_CLOUD_MONITORING_DESTINATION_NODE_NAME,
+			Title:       plan.Title.ValueString(),
+			Description: plan.Description.ValueString(),
+			Inputs:      StringListValueToStringSlice(plan.Inputs),
+			UserConfig: map[string]any{
+				"ack_enabled":      plan.AckEnabled.ValueBool(),
+				"credentials_json": plan.CredentialsJSON.ValueString(),
+				"project_id":       plan.ProjectId.ValueString(),
+				"resource_type":    plan.ResourceType.ValueString(),
+			},
+		},
+	}
+
+	if !plan.ResourceLabels.IsNull() {
+		labelsMap := MapValuesToMapAny(plan.ResourceLabels, &dd)
+		if !dd.HasError() {
+			labelsArray := make([]map[string]string, 0, len(labelsMap))
+			for k, v := range labelsMap {
+				labelsArray = append(labelsArray, map[string]string{"label_name": k, "label_value": v.(string)})
+			}
+			component.UserConfig["resource_labels"] = labelsArray
+		}
+	}
+
+	if previousState != nil {
+		component.Id = previousState.Id.ValueString()
+		component.GenerationId = previousState.GenerationId.ValueInt64()
+	}
+
+	return &component, dd
+}
+
+func GcpCloudMonitoringDestinationToModel(plan *GcpCloudMonitoringDestinationModel, component *Destination) {
+	plan.Id = NewStringValue(component.Id)
+	if component.Title != "" {
+		plan.Title = NewStringValue(component.Title)
+	}
+	if component.Description != "" {
+		plan.Description = NewStringValue(component.Description)
+	}
+	plan.GenerationId = NewInt64Value(component.GenerationId)
+	plan.Inputs = SliceToStringListValue(component.Inputs)
+	plan.AckEnabled = NewBoolValue(component.UserConfig["ack_enabled"].(bool))
+	plan.CredentialsJSON = NewStringValue(component.UserConfig["credentials_json"].(string))
+	plan.ProjectId = NewStringValue(component.UserConfig["project_id"].(string))
+	plan.ResourceType = NewStringValue(component.UserConfig["resource_type"].(string))
+
+	if component.UserConfig["resource_labels"] != nil {
+		labelsArray, _ := component.UserConfig["resource_labels"].([]any)
+		if len(labelsArray) > 0 {
+			labelMap := make(map[string]any, len(labelsArray))
+			for _, obj := range labelsArray {
+				obj := obj.(map[string]any)
+				key := obj["label_name"].(string)
+				value := obj["label_value"].(string)
+				labelMap[key] = value
+			}
+			labelType := GcpCloudMonitoringResourceSchema.Attributes["resource_labels"].GetType().(basetypes.MapType).ElemType
```

Comment:
There's a duplicate `basetypes` import up top.  Use the `.` one by changing this to `MapType` and remove the other import.